spotify_dl/spotify_dl.py

```diff
@@ -8,99 +9,176 @@
+        "-o",
+        "--output",
+        type=str,
+        action="store",
+        help="Specify download directory.",
+        required=False,
+        default=".",
+    )
+    parser.add_argument(
+        "-d",
+        "--download",
+        action="store_true",
+        help="Download using youtube-dl",
+        default=True,
+    )
+    parser.add_argument(
+        "-f",
+        "--format_str",
+        type=str,
+        action="store",
+        help="Specify youtube-dl format string.",
+        default="bestaudio/best",
+    )
+    parser.add_argument(
+        "-k",
+        "--keep_playlist_order",
+        default=False,
+        action="store_true",
+        help="Whether to keep original playlist ordering or not.",
+    )
+    parser.add_argument(
+        "-m",
+        "--skip_mp3",
+        action="store_true",
+        help="Don't convert downloaded songs to mp3",
+    )
+    parser.add_argument(
+        "-s",
+        "--skip_non_music_sections",
+        default=False,
+        action="store_true",
+        help="Whether to skip non-music sections using SponsorBlock API.",
+    )
+    parser.add_argument(
+        "-w",
+        "--no-overwrites",
+        action="store_true",
+        help="Whether we should avoid overwriting the target audio file if it already exists",
+        default=False,
+    )
+    parser.add_argument(
+        "-V",
+        "--verbose",
+        action="store_true",
+        help="Show more information on what" "s happening.",
+    )
+    parser.add_argument(
+        "-v",
+        "--version",
+        action="store_true",
+        help="Shows current version of the program",
+    )
+    parser.add_argument(
+        "-mc",
+        "--multi_core",
+        action="store",
+        type=str,
+        default=0,
+        help="Use multiprocessing [-m [int:numcores]",
+    )
     args = parser.parse_args()
-
+    args.multi_core = int(args.multi_core)
     if args.version:
         console.print(f"spotify_dl [bold green]v{VERSION}[/bold green]")
         sys.exit(0)
 
-    if os.path.isfile(os.path.expanduser('~/.spotify_dl_settings')):
-        with open(os.path.expanduser('~/.spotify_dl_settings')) as file:
-            config = json.loads(file.read())
+    if os.path.isfile(os.path.expanduser("~/.spotify_dl_settings")):
+        with open(os.path.expanduser("~/.spotify_dl_settings")) as file:
+            config = json.load(file)
+            print(config)
 
         for key, value in config.items():
-            if (isinstance(value, bool) and value) or (isinstance(value, str) and value and value.lower() in ['true', 't']):
+            if (isinstance(value, bool) and value) or (
+                isinstance(value, str) and value and value.lower() in ["true", "t"]
+            ):
                 setattr(args, key, True)
             else:
                 setattr(args, key, value)
+    else:
+        print("no config file")
 
     if args.verbose:
         log.setLevel(DEBUG)
-
```

Comment:
the previous check wasnt of help, had to check for the value of the args attr [ url ] not the availability of the attribute
this change served to solve that